chainercv/utils/bbox/bbox_overlap.py

```diff
@@ -0,0 +1,43 @@
+from chainer import cuda
+
+
+def bbox_overlap(bbox_a, bbox_b):
+    """Calculate Jaccard overlap between bounding boxes.
+
+    Jaccard overlap is calculated as a ratio of area of the intersection
+    and area of the union.
+
+    This function accepts both :obj:`numpy.ndarray` and :obj:`cupy.ndarray` as
+    inputs. Please note that both :obj:`bbox_a` and :obj:`bbox_b` need to be
+    same type.
+    The output is same type as the type of the inputs.
+
+    Args:
+        bbox_a (array): An array whose shape is :math:`(N, 4)`.
+            :math:`N` is the number of bounding boxes.
+            The dtype should be :obj:`numpy.float32`.
+        bbox_b (array): An array similar to :obj:`bbox_a`,
+            whose shape is :math:`(K, 4)`.
+            The dtype should be :obj:`numpy.float32`.
+
+    Returns:
+        array:
+        An array whose shape is :math:`(N, K)`. \
+        An element at index :math:`(n, k)` contains overlap between \
+        :math:`n` th bounding box in :obj:`bbox_a` and :math:`k` th bounding \
+        box in :obj:`bbox_b`.
+
+    """
+    if bbox_a.shape[1] != 4 or bbox_b.shape[1] != 4:
+        raise IndexError
+    xp = cuda.get_array_module(bbox_a)
+
+    # left top
+    lt = xp.maximum(bbox_a[:, None, :2], bbox_b[:, :2])
+    # right bottom
+    rb = xp.minimum(bbox_a[:, None, 2:], bbox_b[:, 2:])
```

Comment:
I do not have preference.
Do you?

I like `None` because its shorter. Also, I hesitate to use `xp.newaxis`, which is what I need to use in this case.

But, `np.newaxis` is somewhat more explicit.